### ISSUE TITLE
Exceptions regarding PDO when persisting Magento app and DB

### ISSUE THREAD

**Issue by goncalovf** [bitnami/bitnami-docker-magento#160]
**Description**

I'm experimenting with this image.

First, I tried running the containers without any changes to `docker-compose.yml`. It worked perfectly and I could see the empty Magento at http://127.0.0.1/ as expected.

Then, I tried persisting both `mariadb` and `magento` as explained in the section _Mount host directories as data volumes_ in the `README.md` file. Here is where I encounter my issue. First, after Magento is installed, I get a blank page at `localhost` with the `ERR_EMPTY_RESPONSE` error. Then, after some time, I get 2 exceptions on http://127.0.0.1/ :

```
Exception #0 (Zend_Db_Adapter_Exception): SQLSTATE[HY000] [2002] Connection refused
Exception #1 (PDOException): SQLSTATE[HY000] [2002] Connection refused
```

See the exceptions' stack traces attached: [error-stacktrace.txt](https://github.com/bitnami/bitnami-docker-magento/files/5283806/error-stacktrace.txt)

**Steps to reproduce the issue:**

1. Get Bitnami's `docker-compose.yml`.
2. Only change volumes of both `mariadb` and `magento` services to `- '.:/bitnami'` (lines 11 and 26).
3. Do `docker-compose up -d`.
4. Open `localhost` and see Bitnami's blue screen that says that Magento is installing. Wait for some time (until the installation finishes I think).
5. `localhost` turns blank with `ERR_EMPTY_RESPONSE` error. Wait.
6. `localhost` is blank page with exceptions thrown.

**Describe the results you received:**

- Blank `localhost` with exceptions described above.
- [As expected:] In my project directory I see the `magento` and `mariadb` folders.

**Describe the results you expected:**

- Empty Magento site at `localhost`.

**Additional information you deem important:**

My `docker-compose.yml`:
```
version: '2'
services:
  mariadb:
    image: 'docker.io/bitnami/mariadb:10.3-debian-10'
    environment:
      - ALLOW_EMPTY_PASSWORD=yes
      - MARIADB_USER=bn_magento
      - MARIADB_PASSWORD=magento_db_password
      - MARIADB_DATABASE=bitnami_magento
    volumes:
      - '.:/bitnami'
  magento:
    image: 'docker.io/bitnami/magento:2-debian-10'
    environment:
      - MARIADB_HOST=mariadb
      - MARIADB_PORT_NUMBER=3306
      - MAGENTO_DATABASE_USER=bn_magento
      - MAGENTO_DATABASE_PASSWORD=magento_db_password
      - MAGENTO_DATABASE_NAME=bitnami_magento
      - ELASTICSEARCH_HOST=elasticsearch
      - ELASTICSEARCH_PORT_NUMBER=9200
    ports:
      - '80:80'
      - '443:443'
    volumes:
      - '.:/bitnami'
    depends_on:
      - mariadb
      - elasticsearch
  elasticsearch:
    image: 'docker.io/bitnami/elasticsearch:6-debian-10'
    volumes:
      - 'elasticsearch_data:/bitnami/elasticsearch/data'
volumes:
  elasticsearch_data:
    driver: local
  mariadb_data:
    driver: local
  magento_data:
    driver: local
```

**Version**

- Output of `docker version`:

```
Client: Docker Engine - Community
 Azure integration  0.1.15
 Version:           19.03.12
 API version:       1.40
 Go version:        go1.13.10
 Git commit:        48a66213fe
 Built:             Mon Jun 22 15:43:18 2020
 OS/Arch:           windows/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          19.03.12
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.13.10
  Git commit:       48a66213fe
  Built:            Mon Jun 22 15:49:27 2020
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.2.13
  GitCommit:        7ad184331fa3e55e52b890ea95e65ba581ae3429
 runc:
  Version:          1.0.0-rc10
  GitCommit:        dc9208a3303feef5b3839f4323d9beb36df0a9dd
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683
```

- Output of `docker info`:

```
Client:
 Debug Mode: false

Server:
 Containers: 3
  Running: 3
  Paused: 0
  Stopped: 0
 Images: 3
 Server Version: 19.03.12
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 7ad184331fa3e55e52b890ea95e65ba581ae3429
 runc version: dc9208a3303feef5b3839f4323d9beb36df0a9dd
 init version: fec3683
 Security Options:
  seccomp
   Profile: default
 Kernel Version: 4.19.76-linuxkit
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 2
 Total Memory: 1.945GiB
 Name: docker-desktop
 ID: HCK7:3SK6:Y56C:J3PP:PY7A:PHIH:FNGY:BYPS:5T4M:2YUX:WPEK:ZCMN
 Docker Root Dir: /var/lib/docker
 Debug Mode: true
  File Descriptors: 71
  Goroutines: 79
  System Time: 2020-09-25T12:33:05.90585554Z
  EventsListeners: 5
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
 Product License: Community Engine
```

- Output of `docker-compose version` (if applicable):

```
docker-compose version 1.27.2, build 18f557f9
docker-py version: 4.3.1
CPython version: 3.7.4
OpenSSL version: OpenSSL 1.1.1c  28 May 2019
```

**Additional environment details:**
Docker for Windows 10.

**Comment by goncalovf** [comment]
I changed the issue's content (title and description) because the issue was only temporary it seems. `localhost` only displays the `ERR_EMPTY_RESPONSE` error for some time. However, a new issue is occurring, described above.

**Comment by goncalovf** [comment]
Changed the issue's content once more because I found that the issue was caused by the fact that the DB container was shut down. I guess this happened because I did not have enough RAM allocated to Docker. I started the DB container up again and I found the issue currently in the description above.

Later, I restarted my PC, ran the containers and tried connecting to localhost once more and it works. So I'm closing the issue.
